microprofile/cdi/src/main/java/io/helidon/microprofile/cdi/ContainerInstanceHolder.java

```diff
@@ -32,6 +34,8 @@ final class ContainerInstanceHolder {
     private static final List<Runnable> RESET_LISTENERS = new LinkedList<>();
     private static boolean isReset = false;
 
+    private static final ReentrantReadWriteLock ACCESS_GUARD = new ReentrantReadWriteLock();
```

Comment:
Reading serially, but I'm guessing you won't need the read and write locks, so prefer `ReentrantLock` instead.  Unless I'm wrong. 😄

Reply:
done.